src/js/stories/form-experiment/TextInputHybrid.jsx

```diff
@@ -0,0 +1,117 @@
+import React, { Component, PropTypes } from 'react';
+
+
+const DEFAULT_VALUE = 'value passed before mounting';
+
+
+export default class TextInputHybridStory extends Component {
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            value1: DEFAULT_VALUE,
+            value2: DEFAULT_VALUE,
+        };
+    }
+
+    _onChange(value) {
+        console.log('_onChange:', value);
+    }
+
+    _setValue(value) {
+        console.log('_setValue:', value);
+        this.setState({
+            value2: value,
+        });
+    }
+
+    render() {
+        return (
+            <div style={{ display: 'flex', flexDirection: 'column', alignItems: 'flex-start', height: 300, justifyContent: 'space-between', padding: 10 }}>
+                <h1>Hybrid</h1>
+
+                <span>
+                    <TextInputHybrid onChange={value => this._onChange(value)} />
+                    &nbsp; updates as you type without extra code, without an initial value set
+                </span>
+
+                <span>
+                    <TextInputHybrid onChange={value => this._onChange(value)} value={this.state.value1} />
+                    &nbsp; updates as you type without extra code, with an initial value set
+                </span>
+
+                <br />
+
+                <span>
+                    <button onClick={() => this._setValue(new Date().toString())}>Update 'value' prop</button>
+                    &nbsp; Works fine, since parent has full control over text's state
+                </span>
+
+                <span>
+                    <button onClick={() => this._setValue('')}>Clear 'value' prop</button>
+                    &nbsp; Works fine, since parent has full control over text's state
+                </span>
+
+                <span style={{ background: '#FF9999'}}>
+                    <button onClick={() => this._setValue(DEFAULT_VALUE)}>Reset 'value' prop</button>
+                    &nbsp; Doesn't work after typing, since the value of prop 'value' didn't change
+                </span>
```

Comment:
This illustrates the "unable to reset" bug, which I'm not sure can really be fixed in any reliable way. The workaround would be to clear "value" and then set it to the default value... pretty ugly.